### PR TITLE
Makes composite specification implement Spec

### PR DESCRIPTION
**Pull request by jaapio**
Specification interface is now implemented by composite specification which
makes sense since all implementations are implementing that interface.

### CODE REVIEW

**Comment by jaapio** [comment]
I will start making the build pass after this pr was accepted :-)

File: src/Specification/AndSpecification.php [hunk 1]
```diff
@@ -20,22 +20,22 @@
 final class AndSpecification extends CompositeSpecification implements SpecificationInterface
```
Comment (ashnazg) [inline src/Specification/AndSpecification.php]: Should `implements` be removed here, since it is now inherited from the abstract parent?

Comment (jaapio) [reply src/Specification/AndSpecification.php]: it would be more eager to do so, will remove those

Comment (ashnazg) [reply src/Specification/AndSpecification.php]: Still going to drop the `implements` from the `final` children?

**Comment by jaapio** [comment]
Think this one is ready now.

File: src/Specification/AndSpecification.php (hunk 1 above)
Comment (jaapio) [reply src/Specification/AndSpecification.php]: sorry must have missed this one.

**Review by ashnazg — APPROVED** [review]

**Review by ashnazg — CHANGES_REQUESTED** [review]
ecs didn't like the encompassing parens on that return.  Might as well include that and get the build green.

**Pull request merged**